scripts/eip1559_deployer.ts

```diff
@@ -0,0 +1,457 @@
+import 'hardhat-deploy-ethers';
+import * as fs from 'fs';
+import mainnetData from '../mainnet_deployed.json';
+import testnetData from '../testnet_deployed.json';
+import hre from 'hardhat';
+import { ethers } from 'ethers';
+import '@nomiclabs/hardhat-etherscan/dist/src/type-extensions';
+import { spawnSync } from 'child_process';
+import { InfuraToken } from '../hardhat.config';
+//abi imports
+import postageABI from '../artifacts/src/PostageStamp.sol/PostageStamp.json';
+import redisABI from '../artifacts/src/Redistribution.sol/Redistribution.json';
+import stakingABI from '../artifacts/src/Staking.sol/StakeRegistry.json';
+import oracleABI from '../artifacts/src/PriceOracle.sol/PriceOracle.json';
+import bzzTokenABI from '../artifacts/src/TestToken.sol/TestToken.json';
+
+let account: ethers.Wallet;
+
+interface DeployedContract {
+  abi: Array<unknown>;
+  bytecode: string;
+  address: string;
+  block: number;
+  url: string;
+}
+
+interface DeployedData {
+  chainId: number;
+  networkId: number;
+  contracts: {
+    postageStamp: DeployedContract;
+    redistribution: DeployedContract;
+    staking: DeployedContract;
+    priceOracle: DeployedContract;
+    bzzToken: DeployedContract;
+  };
+}
+
+let configurations: ChainConfig;
+
+interface ChainConfig {
+  chainId?: number;
+  networkId?: number;
+  networkName: string;
+  url: string;
+  deployedData: DeployedData;
+  etherscanKey: string;
+}
+
+const configs: Record<string, ChainConfig> = {
+  testnet: {
+    chainId: hre.network.config.chainId,
+    networkId: 10,
```

Comment:
The value in the `_deployed.json` file should have precedence, see: https://github.com/ethersphere/storage-incentives/blob/master/scripts/deployer.ts#L74
The same for the others.